examples/chip-tool/main.cpp

```diff
@@ -450,19 +460,13 @@ void DoOnOff(DeviceController::ChipDeviceController * controller, Command comman
 #endif
 
     controller->SendMessage(NULL, buffer);
-    // FIXME: waitingForResponse is being written on other threads, presumably.
-    // We probably need some more synchronization here.
-    auto start = std::chrono::system_clock::now();
-    while (waitingForResponse &&
-           std::chrono::duration_cast<std::chrono::minutes>(std::chrono::system_clock::now() - start) < kWaitingForResponseTimeout)
-    {
-        // Just poll for the response.
-        sleep(1);
-    }
 
-    if (waitingForResponse)
+    std::unique_lock<std::mutex> lk(cvWaitingForResponseMutex);
```

Comment:
@bzbarsky-apple will you have a look here?

Reply:
This part looks fine to me, given that we set `waitingForResponse` to true before we call `SendMessage`, so even a sync callback under there will do the right thing.